File: programmers/etc/level2/기능개발.py
def solution(progresses, speeds):
    q = []
    for progress, speed in zip(progresses, speeds):
        if not q or q[-1][0] < -((progress-100)//speed):
            q.append([-((progress-100)//speed), 1])
        else:
            q[-1][1] += 1
    return [i[1] for i in q]


progresses = [93, 30, 55]
speeds = [1, 30, 5]
print(solution(progresses, speeds)) # [2, 1]

progresses = [95, 90, 99, 99, 80, 99]
speeds = [1, 1, 1, 1, 1, 1]
print(solution(progresses, speeds)) # [1, 3, 2]

File: Java/programmers/etc/level2/기능개발.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 기능개발 {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        Queue<Integer> q = new LinkedList<>();
        for (int i=0; i<progresses.length; i++) {
            q.add((int) (Math.ceil((100.0-progresses[i])/speeds[i])));
        }

        List<Integer> answer = new ArrayList<>();

        while (!q.isEmpty()) {
            int day = q.poll();
            int cnt = 1;

            while (!q.isEmpty() && day >= q.peek()) {
                cnt++;
                q.poll();
            }
            answer.add(cnt);
        }

        return answer;
    }
}
